OpenEdXMobile/src/main/java/org/edx/mobile/base/BaseAppActivity.java

```diff
@@ -78,9 +81,19 @@ public boolean showGoogleCastButton() {
 
     @Override
     public void onCastStateChanged(int newState) {
-        if (mediaRouteMenuItem != null) {
-            googleCastDelegate.showIntroductoryOverlay(this, mediaRouteMenuItem);
+        // App initiate `IllegalArgumentException` when showing the IntroductoryOverlay in some cases,
+        // check following issue tracker for more details (still open).
+        // Ref: https://issuetracker.google.com/issues/36191274
+        // TODO: Replace the try-catch block with more appropriate logic / by updating the cast library
```

Comment:
Nits:
- Use a block comment instead of single-line comments.
- Please create a ticket for updating google services specifically mentioning the cast SDK in it.
- Once the ticket is created add an item in its acceptance criteria to revisit this part of code in it.
- Also the add the ticket's link to this TODO item.

Reply:
done.